Sources/TuistGenerator/Models/Settings.swift

```diff
@@ -29,27 +23,43 @@ public class Configuration: Equatable {
 }
 
 public class Settings: Equatable {
+    public static let `default` = Settings(configurations: [.release: nil, .debug: nil])
+
     // MARK: - Attributes
 
     public let base: [String: String]
-    public let debug: Configuration?
-    public let release: Configuration?
+    public let configurations: [BuildConfiguration: Configuration?]
 
     // MARK: - Init
 
     public init(base: [String: String] = [:],
-                debug: Configuration?,
-                release: Configuration?) {
+                configurations: [BuildConfiguration: Configuration?]) {
         self.base = base
-        self.debug = debug
-        self.release = release
+        self.configurations = configurations
+    }
+
+    // MARK: - Internal
+
+    static func ordered(configurations: [BuildConfiguration: Configuration?])
```

Comment:
Nitpick: What about moving this one into an extension to hashes whose key is a `BuildConfiguration` and the value is a `Configuration`?

Reply:
Same comment applies to the two methods below 😛. It'll be easier to unit-test them.

Reply:
Oh, right! Extension seems to fit a way better 👍. I wasn't sure where to put the tests for the extension methods, and they ended up inside `SettingsTests` suite. Would you prefer to use `DictionaryBuildConfigurationExtensionTests` or something similar?